runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -133,6 +133,13 @@ function LanguageTree:parse()
 
   for lang, injection_ranges in pairs(injections_by_lang) do
     local has_lang = language.require_language(lang, nil, true)
+    local lang_by_extension = require'vim.filetype'._ft_by_extension(lang)
```

Comment:
totally a hard-coded hack. just a demo what could be done. this could go into `_get_injections` for a `@extension` capture.